### ISSUE TITLE
Bug: [no-dynamic-delete] Not fixable

### ISSUE THREAD

**Issue by mho22** [typescript-eslint/typescript-eslint#11872]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=MYewdgzgLgBKZQIYEswFMBOAuGBvGA2sjtBqgOYC6OYArgLYBGmMAvjALx4wDkAZiBA8cAFgBMbANwAoACZoANmiho44JKkwF%2BgnpUlA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNCmBDALaUyPREnyJ0URNGgdokcGAC%2BIFUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const container: { [i: string]: number } = { 'foo': 42 };
delete container['foo'];
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/no-dynamic-delete": "error"
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

```typescript
const container: { [i: string]: number } = { 'foo': 42 };
delete container.foo;
```

### Actual Result

```typescript
const container: { [i: string]: number } = { 'foo': 42 };
delete container['foo'];
```

### Additional Info

The `no-dynamic-delete` rule that is described as `fixable` doesn't look like it is.

**Comment by kirkwaiblinger** [comment]
Hi @mho22!

It's intentional that this snippet isn't flagged. Since a literal string is provided as the member key, there's no "dynamic" behavior, and the code is equivalent to `delete container.foo`. See also the [rule examples](https://typescript-eslint.io/rules/no-dynamic-delete/#examples) where this is explicitly documented.

For a rule that lets you control the style of member access, see https://typescript-eslint.io/rules/dot-notation/.

**Comment by JoshuaKGoldberg** [comment]
Oh, hang on, @kirkwaiblinger I don't figure out any cases where the fixer comes into play. The implementation does include it but none of the tests in `no-dynamic-delete.test.ts` have a filled-out `output` - just `null`s. Can you find a way to get the fixer to trigger?

(I was just poking around here and planned on responding soon. Race condition!)

**Comment by kirkwaiblinger** [comment]
Oh - also possible there is a bug! Took the issue to be about that snippet (which is working as intended) rather than the rule generally

**Comment by kirkwaiblinger** [comment]
It looks like the rule used to autofix `delete a['foo']` to `delete a.foo` but this was changed in https://github.com/typescript-eslint/typescript-eslint/pull/9280. Ostensibly the fixer logic is now dead and can be removed I'm guessing

**Comment by JoshuaKGoldberg** [comment]
Nice. Accepting PRs!

**Comment by mho22** [comment]
@kirkwaiblinger Wow, I didn’t expect this issue to be handled so quickly. I don’t know if you want to suggest a pull request, but I can propose one if you’d like. You have more expertise here, so I’ll wait for your input.

**Comment by kirkwaiblinger** [comment]
@mho22 a pull request is welcome! See https://typescript-eslint.io/contributing/pull-requests

**Comment by mho22** [comment]
@kirkwaiblinger @JoshuaKGoldberg [Pull request](#11876) ready for review! I don't think documentation needs to be modified.
